apis/offloading/v1alpha1/namespaceoffloading_types.go

```diff
@@ -0,0 +1,162 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GlobalOffloadingPhase represents namespaces offloading status.
+type GlobalOffloadingPhase string
+
+const (
+	// NamespaceOffloadingCompleted -> remote Namespaces have been correctly created on previously selected clusters.
+	NamespaceOffloadingCompleted GlobalOffloadingPhase = "OffloadingCompleted"
+	// NamespaceOffloadingNotConfigured -> user has not entered the liqo.io/enabled="true" label yet.
+	NamespaceOffloadingNotConfigured GlobalOffloadingPhase = "OffloadingNotConfigured"
```

Comment:
Why is it required to manually add the label? In kubernetes, you cannot assume operations are performed in a given order (e.g. you first create the resource and then add the label), so this would lead to race conditions if the label triggers the automatic creation of a default NamespaceOffloading resource.

Reply:
yes you are right